ts/session/disappearing_messages/index.ts

```diff
@@ -543,18 +544,40 @@ function getMessageReadyToDisappear(
     messageExpirationFromRetrieve &&
     messageExpirationFromRetrieve > 0
   ) {
-    const expirationStartTimestamp = messageExpirationFromRetrieve - expireTimer * 1000;
-    const expires_at = messageExpirationFromRetrieve;
-    // TODO a message might be added even when it expired, but the period cleaning of expired message will pick it up and remove it soon enough
-    window.log.debug(
-      `incoming DaR message already read by another device, forcing readAt ${(Date.now() -
-        expirationStartTimestamp) /
-        1000}s ago, so with ${(expires_at - Date.now()) / 1000}s left`
-    );
-    messageModel.set({
-      expirationStartTimestamp,
-      expires_at,
-    });
+    /**
+     * Edge case: when we send a message before we poll for a message sent earlier, our convo volatile update will
+     * mark that incoming message as read right away (because it was sent earlier than our latest convolatile lastRead).
+     * To take care of this case, we need to check if the expiration of an incoming DaR, alreadt marked as read message looks to not have been updated yet.
```

Comment:
`To take care of this case, we need to check if the expiration of an incoming DaR, alreadt marked as read message looks to not have been updated yet.` -> `To take care of this case, we need to check if the expiration of an incoming DaR is already marked as read and looks like it has not been updated yet.`